### ISSUE TITLE
[Vitest] Survived mutants for use cases covered on the tests 

### ISSUE THREAD

**Issue by islasjuanp** [stryker-mutator/stryker-js#4306]
Hi everyone 👋 

**Summary**

I am detecting some survival mutants for use cases that are covered in the tests. It means that if I make the same code change, some tests fail due to the change. 

I will provide a small example of it: 

```
[Survived] StringLiteral
packages/functions/src/auth/preSignUp.ts:25:19
-     throw new Error('Unauthorized to create account')
+     throw new Error("")
```

Test that covered that use case: 
```
AssertionError: expected Error to deeply equal Error: Unauthorized to create account

  - Expected  - 0
  + Received  + 1

    Error {
  -   message: 'Unauthorized to create account',
  +   message: '',
    }
```

**Stryker config**

```json
{
  "$schema": "./node_modules/@stryker-mutator/core/schema/stryker-schema.json",
  "_comment": "This config was generated using 'stryker init'. Please take a look at: https://stryker-mutator.io/docs/stryker-js/configuration/ for more information.",
  "packageManager": "npm",
  "mutate": [ 
    "packages/**/*.ts", 
    "!packages/**/node_modules/**/*.ts", 
    "!packages/**/test/**/*.ts"
  ],
  "reporters": [
    "html",
    "clear-text",
    "progress"
  ],
  "testRunner": "vitest",
  "vitest": {
    "configFile": "vite.config.ts"
  },
  "coverageAnalysis": "off",
  "thresholds": {
    "high": 80,
    "low": 60,
    "break": 40
  }
}
```

**Test runner config**

**Stryker environment**

```
$ npm ls | grep stryker
├── @stryker-mutator/core@7.0.2
├── @stryker-mutator/vitest-runner@7.0.2
```

```
+-- vitest@0.31.4
+-- sst@2.11.17
```

**Your Environment**

| software         | version(s)
| ---------------- | -------
| node             | v18.12.1
| npm              |  8.19.2
| Operating System | macOS Ventura

**Comment by islasjuanp** [comment]
Thanks, @nicojs 

I've just added a small example of this issue on the following public repository: https://github.com/islasjuanp/api-auth-jwt-cognito-user-pool

**Comment by nicojs** [comment]
Sorry, but I can't use your example:

```shell
npm i
npm test
```

Results in:

```
nicojs@NicoJ01 ~/github/api-auth-jwt-cognito-user-pool (main)$ npm run test

> api-auth-jwt-cognito-user-pool@0.0.0 test
> vitest run


 RUN  v0.31.4 /home/nicojs/github/api-auth-jwt-cognito-user-pool

 ✓ packages/functions/test/auth/preSignUp.spec.ts (6)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Errors ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

Vitest caught 2 unhandled errors during the test run.
This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error: promise resolved "[object Object]" instead of rejecting
 ❯ node_modules/@vitest/expect/dist/index.js:1261:21

This error originated in "packages/functions/test/auth/preSignUp.spec.ts" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error: promise resolved "[object Object]" instead of rejecting
 ❯ node_modules/@vitest/expect/dist/index.js:1261:21

This error originated in "packages/functions/test/auth/preSignUp.spec.ts" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 Test Files  1 passed (1)
      Tests  6 passed (6)
     Errors  2 errors
   Start at  13:47:52
   Duration  856ms (transform 40ms, setup 0ms, collect 30ms, tests 3ms, environment 0ms, prepare 60ms)
```

**Comment by nicojs** [comment]
I'm pretty sure the mutant survived because you forgot to `await` on `expect(...).rejected....`. See my PR that fixes it:

```diff
- expect(handler(event, context)).rejects.toEqual(
+ await expect(handler(event, context)).rejects.toEqual(
   new Error('Unauthorized to create account')
 )
```

 https://github.com/islasjuanp/api-auth-jwt-cognito-user-pool/pull/1

You can also see this if you run vitest with the mutant active (without `await`). It reports a successful run (all tests pass), but then reports the error anyway:  
 
![image](https://github.com/stryker-mutator/stryker-js/assets/1828233/805c3ad2-8cde-427c-b558-22abc91578ac)

We might be able to battleharden the vitest runner better to detect forgotten awaits 🤔

**Comment by islasjuanp** [comment]
Thanks @nicojs, It would be really helpful because it is hard to understand what is happening otherwise 😄

**Comment by nicojs** [comment]
Released in 7.1
